Logger/BaseEntity.cs

```diff
@@ -0,0 +1,8 @@
+namespace Logger;
+
+public abstract record class BaseEntity : IEntity
+{
+    public Guid Id { get; init; } = Guid.NewGuid();
```

Comment:
Is there ever a time the Id is set by the calling class? 
```suggestion
    public Guid Id { get; } = Guid.NewGuid();
```